core/src/main/java/io/questdb/griffin/engine/functions/groupby/FirstStrGroupByFunction.java

```diff
@@ -100,12 +140,18 @@ public boolean isScalar() {
     @Override
     public void pushValueTypes(ArrayColumnTypes columnTypes) {
         this.valueIndex = columnTypes.getColumnCount();
-        columnTypes.add(ColumnType.BOOLEAN);
+        columnTypes.add(ColumnType.INT); // sink index
+        columnTypes.add(ColumnType.BOOLEAN); // null flag
     }
 
     @Override
     public void setNull(MapValue mapValue) {
-        mapValue.putBool(valueIndex, false);
-        sink.clear();
+        mapValue.putBool(valueIndex + 1, true);
```

Comment:
Could you explain what does this do? I couldn't hit this with the tests from `FirstAndLastStrGroupByFunctionFactoryTest`

Reply:
It gets hit in latest commit, once I added fill(prev) to sample by tests.